wolfcrypt/src/evp.c

```diff
@@ -3143,6 +3143,69 @@ int wolfSSL_PKCS5_PBKDF2_HMAC(const char *pass, int passlen,
 }
 #endif /* !NO_PWDBASED */
 
+
+#if defined(HAVE_SCRYPT) && defined(HAVE_PBKDF2) && !defined(NO_PWDBASED) && \
+                                                    !defined(NO_SHA)
+/**
+ * Derives a key from the specified password and the salt using SCRYPT 
+ * algorithm.
+ * 
+ * Parameters:
+ * - pass      :password data. no need to be null-terminated. NULL is accepted.
+ * - passlen   :lengh of the password. Must be 0 when pass is NULL.
+ * - salt      :salt. NULL is accepted.
+ * - saltlen   :length of the salt. Must be 0 when salt is NULL.
+ * - N         :cost parameter. Must be grater or equal to 2 and be a power of 2.     
+ * - r         :block size. Must 1 or greater.
+ * - p         :parallelism 
+ * - maxmem    :maximum size of buffer used for calculation in definition,
+ *              Not reffered in this implementation.
+ * - key       :derived key.
+ * - keylen    :length of the derived key
+ * 
+ * Returns:
+ *   1 on success, otherwize 0.
+ */
+int wolfSSL_EVP_PBE_scrypt(const char *pass, size_t passlen,
+                            const unsigned char *salt, size_t saltlen,
+                            uint64_t N, uint64_t r, uint64_t p, 
+                            uint64_t maxmem, unsigned char *key, size_t keylen)
+{
+    (void)maxmem;
+    int ret;
+    int exp = 0;
+
+    if (N < 2 || (N & (N-1))  || r <= 0 || p <= 0 )
```

Comment:
What is `(N & (N-1)` checking. Please add comment. Also adjust the extra space.

Reply:
The check is to test that N is not a power of 2.  We need to test N is a power of 2  and is grater than one in here.